provider/pkg/provider/provider.go

```diff
@@ -1654,6 +1654,10 @@ func (k *kubeProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*p
 				switch v.Kind {
 				case pulumirpc.PropertyDiff_ADD_REPLACE, pulumirpc.PropertyDiff_DELETE_REPLACE, pulumirpc.PropertyDiff_UPDATE_REPLACE:
 					replaces = append(replaces, k)
+				case pulumirpc.PropertyDiff_DELETE:
+					if k == "metadata" {
```

Comment:
Is it ever a possibility that `metadata` remains but `metadata.name` is removed?